### PR TITLE
Create compliance test runner

### PR DESCRIPTION
**Pull request by temyers**
Creates compliance runner for client use.

Considerations:
Opted to create a separate client class, rather than a lambda for consideration when loading specs, etc.  Options considered:
1. Lambda function loads specs passed in via the event, optionally including extra parameters for the suite
2. Suites passed as a callback to the Class constructor
3. *selected option* - client loads the suites after construction, but before execution.

Option 1 discounted, since when thinking from a security standpoint, this would allow for potentially arbitrary code execution within the compliance tests - since limited semantic validation could be applied to the loading of functions.

Option 2 discounted, since it would make initialising the suites more cumbersome.

Option 3, as demonstrated in the README.md, results in concise API for the client, whilst providing the flexibility for initialising the compliance suites.

### CODE REVIEW

File: src/runners/JasmineComplianceRunner.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const TSConsoleReporter = require('jasmine-ts-console-reporter');
+
+export const lambdaNotificationReporter = (callback: Callback) => {
+  return {
+    results : [],
+    testErrors: [],
+    passedExpectations : [],
+    jasmineDone: function (result) {
+      if (result.overallStatus === 'failed') {
+        let failedTests = this.testErrors.map(x => x.fullName).join(',');
```
Comment (raf-vgw) [inline src/runners/JasmineComplianceRunner.ts]: Minor: join with ', ' (with a space). I forgot this when I originally wrote it.

File: src/runners/JasmineComplianceRunner.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    env.specFilter = () => true
+    let reporter = new TSConsoleReporter()
+    reporter.setOptions( {showColor : false})
+    env.addReporter(reporter);
+    //  env.throwOnExpectationFailure(true)
+  
+}
+
+export class JasmineComplianceRunner {
+    jasmine
```
Comment (raf-vgw) [inline src/runners/JasmineComplianceRunner.ts]: should this be `let jasmine = undefined`?

Comment (temyers) [reply src/runners/JasmineComplianceRunner.ts]: Not required - the above is declaring it as an instance variable (AIUI)
Adding the let is a compile error.

File: src/runners/JasmineComplianceRunner.ts (hunk 1 above)
Comment (temyers) [reply src/runners/JasmineComplianceRunner.ts]: ok - will fix.

**Pull request merged**